lib/modal-box/index.d.ts

```diff
@@ -1,96 +1,97 @@
-         * If the validation function return true, the modal box closes, otherwise it stays open
-         * @param button
-         */
-        validation?: (button: BUTTON) => boolean;
-        /**
-         * Specify the title of the modal box
-         */
-        title?: string;
-        /**
-         * Specify the content that you wish to put inside the modal box
-         */
-        body?: HTMLElement;
-    }
+    wrapper: HTMLElement;
     /**
-     * Open a modal box with the given content
-     * @param content The content to display, as an HTMLElement
-     * @param options The {@link Options} to use for this modal box
-     * @returns {{modalBox: (HTMLDivElement|HTMLElement), overlay: (HTMLDivElement|HTMLElement), wrapper: (HTMLDivElement|HTMLElement), buttons: HTMLElement, content: HTMLElement, close: (function(BUTTON=, boolean=): (boolean|boolean))}}
+     * The availables buttons (Ok, Apply, Cancel, etc.)
      */
-    function open(content: HTMLElement, options?: Options): ModalBox;
+    buttons: HTMLElement;
     /**
-     * Close all open modal box instance
-     * @param forceClose Will skip validation
+     * The content itself
      */
-    function close(forceClose?: boolean): void;
+    content: HTMLElement;
+    /**
+     * The function that can be called to close the modal box. Note that this is also called by validation button such as APPLY, YES, etc.<br/>
+     * Force close will close all open modalbox and skip the validation (if one was provided)
+     * @param button
+     * @param forceClose
+     */
+    close: (button?: BUTTON, forceClose?: boolean) => boolean;
+    open: (content: HTMLElement, options?: Options) => ModalBox;
+  }
+  /**
+   * Possible options when creating a ModalBox
+   */
+  interface Options {
+    /**
+     * Specify if you wish to open the modal box full screen. Default is `false`. If false, the modal box will fit the size of the content.
+     */
+    fullscreen?: boolean;
+    /**
+     * Specify that you wish the modal box to close when the user click on the title. Default is `false`.
+     */
+    titleClose?: boolean;
+    /**
+     * Specify if you wish to close the modal box when the overlay (black background) is clicked. Default is `false`.
+     */
+    overlayClose?: boolean;
+    /**
+     * Specify that you wish to add a prefix to the class name of the modal box container, to not clash with existing css in the page
+     */
+    className?: string;
+    /**
+     * The button you wish to create (Using {@link BUTTON} enum
+     */
+    buttons?: number;
+    /**
+     * Specify a validation function, which receives the button that was pressed.<br/>
+     * If the validation function return true, the modal box closes, otherwise it stays open
+     * @param button
+     */
+    validation?: (button: BUTTON) => boolean;
+    /**
+     * Specify the title of the modal box
+     */
+    title?: string | HTMLElement;
```

Comment:
This is one of the changes I made. I changed this type from `string` to `string | HTMLElement` since some old code was already setting an `HTMLElement` on this property.